### PR TITLE
UCP/AM: Reduce req->msg_proto size

### PR DESCRIPTION
**Pull request by brminich**
## What
Reduce req->send.msg_proto.am size

## Why ?
Free 8 bytes in req->send.msg_proto for adding req_id

### CODE REVIEW

File: src/ucp/core/ucp_am.c [hunk 1]
```diff
@@ -280,6 +280,7 @@ ucp_am_zcopy_pack_user_header(ucp_request_t *req)
 
     ucp_am_pack_user_header(reg_desc + 1, req);
     req->send.msg_proto.am.reg_desc = reg_desc;
+    req->flags                     |= UCP_REQUEST_FLAG_AM_HDR_PACKED;
```
Comment (yosefe) [inline src/ucp/core/ucp_am.c:283]: this phase should be done when AM zcopy send is started, not during progress. then, we won't need to save *header in the request at all

Comment (brminich) [reply src/ucp/core/ucp_am.c:283]: ok, but header is needed for bcopy and short

Comment (yosefe) [reply src/ucp/core/ucp_am.c:283]: right, still need union, but each proto would use other field of it (can comment this in the struct)

**Review by yosefe — APPROVED** [review]

**Pull request merged**